{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Schoolar - Sprint 2**\n",
    "\n",
    "## **Objetivo**\n",
    "Corrigir uma prova pelo Chat-GPT\n",
    "\n",
    "## **Descrição do Desafio**\n",
    "Este é um projeto em Python que utiliza as bibliotecas EasyOCR e OpenAI para digitalizar textos em imagens de provas, corrigir questões dissertativas e retornar a resposta corrigida.\n",
    "\n",
    "Primeiro, a imagem da prova é lida com a biblioteca OpenCV e é separada em duas partes, a questão e a resposta. Em seguida, a biblioteca EasyOCR é usada para digitalizar o texto da questão e da resposta.\n",
    "\n",
    "Depois disso, a biblioteca OpenAI é utilizada para corrigir a questão. É criado um prompt que é enviado para a API da OpenAI com a questão e a resposta digitalizadas como entrada. A API retorna uma resposta corrigida em uma escala de 0 a 10 e a certeza de sua correção em uma escala de 0 a 1.\n",
    "\n",
    "Finalmente, a resposta corrigida é retornada como resultado. Este projeto pode ser útil para estudantes ou professores que precisam corrigir muitas questões dissertativas de maneira rápida e automatizada."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### **Roteiro de Desenvolvimento**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1- Instale a biblioteca"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: easyocr in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (1.6.2)Note: you may need to restart the kernel to use updated packages.\n",
      "\n",
      "Requirement already satisfied: python-bidi in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (0.4.2)\n",
      "Requirement already satisfied: Shapely in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (2.0.1)\n",
      "Requirement already satisfied: ninja in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (1.11.1)\n",
      "Requirement already satisfied: torch in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (2.0.0)\n",
      "Requirement already satisfied: torchvision>=0.5 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (0.15.1)\n",
      "Requirement already satisfied: Pillow in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (9.5.0)\n",
      "Requirement already satisfied: pyclipper in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (1.3.0.post4)\n",
      "Requirement already satisfied: numpy in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (1.23.5)\n",
      "Requirement already satisfied: scikit-image in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (0.20.0)\n",
      "Requirement already satisfied: scipy in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (1.9.1)\n",
      "Requirement already satisfied: opencv-python-headless<=4.5.4.60 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (4.5.4.60)\n",
      "Requirement already satisfied: PyYAML in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from easyocr) (6.0)\n",
      "Requirement already satisfied: requests in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from torchvision>=0.5->easyocr) (2.28.1)\n",
      "Requirement already satisfied: typing-extensions in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from torch->easyocr) (4.5.0)\n",
      "Requirement already satisfied: filelock in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from torch->easyocr) (3.12.0)\n",
      "Requirement already satisfied: networkx in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from torch->easyocr) (3.1)\n",
      "Requirement already satisfied: sympy in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from torch->easyocr) (1.11.1)\n",
      "Requirement already satisfied: jinja2 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from torch->easyocr) (3.1.2)\n",
      "Requirement already satisfied: six in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from python-bidi->easyocr) (1.16.0)\n",
      "Requirement already satisfied: PyWavelets>=1.1.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from scikit-image->easyocr) (1.4.1)\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from scikit-image->easyocr) (23.0)\n",
      "Requirement already satisfied: lazy_loader>=0.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from scikit-image->easyocr) (0.2)\n",
      "Requirement already satisfied: imageio>=2.4.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from scikit-image->easyocr) (2.28.1)\n",
      "Requirement already satisfied: tifffile>=2019.7.26 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from scikit-image->easyocr) (2023.4.12)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from jinja2->torch->easyocr) (2.1.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests->torchvision>=0.5->easyocr) (3.4)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests->torchvision>=0.5->easyocr) (2.1.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests->torchvision>=0.5->easyocr) (2022.9.24)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests->torchvision>=0.5->easyocr) (1.26.12)\n",
      "Requirement already satisfied: mpmath>=0.19 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from sympy->torch->easyocr) (1.3.0)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: openai in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (0.27.6)\n",
      "Requirement already satisfied: tqdm in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from openai) (4.46.0)\n",
      "Requirement already satisfied: requests>=2.20 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from openai) (2.28.1)\n",
      "Requirement already satisfied: aiohttp in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from openai) (3.8.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.20->openai) (3.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.20->openai) (2022.9.24)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.20->openai) (1.26.12)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from requests>=2.20->openai) (2.1.1)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (1.3.1)\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (4.0.2)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (6.0.4)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (1.3.3)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (1.9.2)\n",
      "Requirement already satisfied: attrs>=17.3.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from aiohttp->openai) (23.1.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: matplotlib in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (3.7.1)Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Requirement already satisfied: inline in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (0.0.1)\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from matplotlib) (23.0)\n",
      "Requirement already satisfied: pyparsing>=2.3.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (3.0.9)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (9.5.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (1.0.7)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (1.4.4)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (0.11.0)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from matplotlib) (2.8.2)\n",
      "Requirement already satisfied: importlib-resources>=3.2.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (5.12.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (4.39.3)\n",
      "Requirement already satisfied: numpy>=1.20 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib) (1.23.5)\n",
      "Requirement already satisfied: zipp>=3.1.0 in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from importlib-resources>=3.2.0->matplotlib) (3.15.0)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\rafae\\appdata\\roaming\\python\\python39\\site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)\n",
      "Requirement already satisfied: opencv-python in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (4.7.0.72)Note: you may need to restart the kernel to use updated packages.\n",
      "\n",
      "Requirement already satisfied: numpy>=1.17.0 in c:\\users\\rafae\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from opencv-python) (1.23.5)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the 'c:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "%pip install easyocr\n",
    "%pip install openai\n",
    "%pip install matplotlib inline\n",
    "%pip install opencv-python"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2 - Ler a imagem"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "partially initialized module 'cv2' has no attribute 'gapi_wip_gst_GStreamerPipeline' (most likely due to a circular import)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[2], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mmatplotlib\u001b[39;00m \u001b[39mimport\u001b[39;00m pyplot \u001b[39mas\u001b[39;00m plt\n\u001b[1;32m----> 2\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mcv2\u001b[39;00m\n\u001b[0;32m      4\u001b[0m \u001b[39m# Ler a prova\u001b[39;00m\n\u001b[0;32m      5\u001b[0m prova \u001b[39m=\u001b[39m cv2\u001b[39m.\u001b[39mimread(\u001b[39m\"\u001b[39m\u001b[39m./prova.png\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\cv2\\__init__.py:181\u001b[0m\n\u001b[0;32m    176\u001b[0m             \u001b[39mif\u001b[39;00m DEBUG: \u001b[39mprint\u001b[39m(\u001b[39m\"\u001b[39m\u001b[39mExtra Python code for\u001b[39m\u001b[39m\"\u001b[39m, submodule, \u001b[39m\"\u001b[39m\u001b[39mis loaded\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m    178\u001b[0m     \u001b[39mif\u001b[39;00m DEBUG: \u001b[39mprint\u001b[39m(\u001b[39m'\u001b[39m\u001b[39mOpenCV loader: DONE\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m--> 181\u001b[0m bootstrap()\n",
      "File \u001b[1;32mc:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\cv2\\__init__.py:175\u001b[0m, in \u001b[0;36mbootstrap\u001b[1;34m()\u001b[0m\n\u001b[0;32m    172\u001b[0m \u001b[39mif\u001b[39;00m DEBUG: \u001b[39mprint\u001b[39m(\u001b[39m'\u001b[39m\u001b[39mOpenCV loader: binary extension... OK\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m    174\u001b[0m \u001b[39mfor\u001b[39;00m submodule \u001b[39min\u001b[39;00m __collect_extra_submodules(DEBUG):\n\u001b[1;32m--> 175\u001b[0m     \u001b[39mif\u001b[39;00m __load_extra_py_code_for_module(\u001b[39m\"\u001b[39;49m\u001b[39mcv2\u001b[39;49m\u001b[39m\"\u001b[39;49m, submodule, DEBUG):\n\u001b[0;32m    176\u001b[0m         \u001b[39mif\u001b[39;00m DEBUG: \u001b[39mprint\u001b[39m(\u001b[39m\"\u001b[39m\u001b[39mExtra Python code for\u001b[39m\u001b[39m\"\u001b[39m, submodule, \u001b[39m\"\u001b[39m\u001b[39mis loaded\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m    178\u001b[0m \u001b[39mif\u001b[39;00m DEBUG: \u001b[39mprint\u001b[39m(\u001b[39m'\u001b[39m\u001b[39mOpenCV loader: DONE\u001b[39m\u001b[39m'\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\cv2\\__init__.py:28\u001b[0m, in \u001b[0;36m__load_extra_py_code_for_module\u001b[1;34m(base, name, enable_debug_print)\u001b[0m\n\u001b[0;32m     26\u001b[0m native_module \u001b[39m=\u001b[39m sys\u001b[39m.\u001b[39mmodules\u001b[39m.\u001b[39mpop(module_name, \u001b[39mNone\u001b[39;00m)\n\u001b[0;32m     27\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m---> 28\u001b[0m     py_module \u001b[39m=\u001b[39m importlib\u001b[39m.\u001b[39;49mimport_module(module_name)\n\u001b[0;32m     29\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mImportError\u001b[39;00m \u001b[39mas\u001b[39;00m err:\n\u001b[0;32m     30\u001b[0m     \u001b[39mif\u001b[39;00m enable_debug_print:\n",
      "File \u001b[1;32mc:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\lib\\importlib\\__init__.py:127\u001b[0m, in \u001b[0;36mimport_module\u001b[1;34m(name, package)\u001b[0m\n\u001b[0;32m    125\u001b[0m             \u001b[39mbreak\u001b[39;00m\n\u001b[0;32m    126\u001b[0m         level \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m \u001b[39m1\u001b[39m\n\u001b[1;32m--> 127\u001b[0m \u001b[39mreturn\u001b[39;00m _bootstrap\u001b[39m.\u001b[39;49m_gcd_import(name[level:], package, level)\n",
      "File \u001b[1;32mc:\\Users\\rafae\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\cv2\\gapi\\__init__.py:301\u001b[0m\n\u001b[0;32m    296\u001b[0m         \u001b[39mreturn\u001b[39;00m \u001b[39mcls\u001b[39m\n\u001b[0;32m    298\u001b[0m     \u001b[39mreturn\u001b[39;00m kernel_with_params\n\u001b[1;32m--> 301\u001b[0m cv\u001b[39m.\u001b[39mgapi\u001b[39m.\u001b[39mwip\u001b[39m.\u001b[39mGStreamerPipeline \u001b[39m=\u001b[39m cv\u001b[39m.\u001b[39;49mgapi_wip_gst_GStreamerPipeline\n",
      "\u001b[1;31mAttributeError\u001b[0m: partially initialized module 'cv2' has no attribute 'gapi_wip_gst_GStreamerPipeline' (most likely due to a circular import)"
     ]
    }
   ],
   "source": [
    "from matplotlib import pyplot as plt\n",
    "import cv2\n",
    "\n",
    "# Ler a prova\n",
    "prova = cv2.imread(\"./prova.png\")\n",
    "\n",
    "plt.figure(figsize =(16,16))\n",
    "plt.imshow(prova, interpolation=\"none\")\n",
    "plt.show()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3 - Separando Questão e Resposta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Separando a questão\n",
    "questao = prova[0:280,0:700]\n",
    "plt.figure(figsize =(16,16))\n",
    "plt.imshow(questao, interpolation=\"none\", cmap=\"gray\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Separando a resposta\n",
    "resposta = prova[280:420,0:700]\n",
    "plt.figure(figsize =(16,16))\n",
    "plt.imshow(resposta, interpolation=\"none\", cmap=\"gray\")\n",
    "plt.show()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "4 - Criar função para digitalizar a prova"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import easyocr\n",
    "\n",
    "def digitalizar_texto(img):\n",
    "    # Criar um leitor OCR em português\n",
    "    reader = easyocr.Reader(['pt'])\n",
    "\n",
    "    # Realizar OCR na imagem\n",
    "    texto = reader.readtext(img, detail=0, paragraph=True)\n",
    "\n",
    "    return ' '.join(texto)\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "5 - Digitalizar a questão e a resposta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Questao\n",
    "questao_digitalizada = digitalizar_texto(questao)\n",
    "print(questao_digitalizada)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Resposta\n",
    "resposta_digitalizada = digitalizar_texto(resposta)\n",
    "print(resposta_digitalizada)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "6 - Criando a função de integração do ChatGPT para correção de provas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "GPTkey =  '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'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai\n",
    "\n",
    "def corrigir_questao(questao, resposta):\n",
    "    openai.api_key = GPTkey\n",
    "\n",
    "    prompt = f\"Vou utilizar voce para correção de uma questao e quero que voce de uma nota de 0 a 10: Questão: {questao}. Resposta: {resposta}\"\n",
    "\n",
    "    response = openai.Completion.create(\n",
    "        model=\"text-davinci-003\",\n",
    "        prompt=prompt,\n",
    "        max_tokens=20,\n",
    "        n=1,\n",
    "        stop=None,\n",
    "        temperature=0.5,\n",
    "    )\n",
    "\n",
    "    resposta = response.choices[0].text.strip()\n",
    "\n",
    "    return resposta\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "7 - Corrigindo a prova"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resultado = corrigir_questao(questao_digitalizada, resposta_digitalizada)\n",
    "print(resultado)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
